pandas/_libs/tslibs/nattype.pyx

```diff
@@ -123,7 +123,9 @@ cdef class _NaT(datetime):
             return c_NaT
         elif getattr(other, '_typ', None) in ['dateoffset', 'series',
                                               'period', 'datetimeindex',
-                                              'timedeltaindex']:
+                                              'datetimearray',
```

Comment:
this list and the one below are different, why is that? is this one missing periodindex?

Reply:
It's definitely weird, and this is hiding other bugs e.g. with ndarray or str.  I've got a separate branch to fix this Once And For All that removes these lists entirely.  For now I want to fix this particular bug so we can simplify the tar out of core.ops